# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1262849342 -3600
#      Thu Jan 07 08:29:02 2010 +0100
# Node ID c670c8be7f938bb75599f0feda0496250d231b08
# Parent  ca93f583573dbacf72c09071cd40fc3b5bef9380
fix incorrect octave_interrupt_state escaped from liboctave

diff --git a/libcruft/ChangeLog b/libcruft/ChangeLog
--- a/libcruft/ChangeLog
+++ b/libcruft/ChangeLog
@@ -1,8 +1,13 @@
+2010-01-07  Jaroslav Hajek  <highegg@gmail.com>
+
+	* misc/quit.cc (octave_rethrow_exception): Set octave_interrupt_state
+	to -1 when throwing octave_interrupt_exception.
+
 2010-01-05  John W. Eaton  <jwe@octave.org>
 
 	* Makefile.am (AM_CPPFLAGS): New variable.
 
 2009-12-31  Rik <octave@nomad.inbox5.com>
 
 	* Makefile.am: Reverse using simple_move_if_change_rule for cruft.def
 	which interferes with make timestamp algorithm.
diff --git a/libcruft/misc/quit.cc b/libcruft/misc/quit.cc
--- a/libcruft/misc/quit.cc
+++ b/libcruft/misc/quit.cc
@@ -77,17 +77,20 @@ octave_throw_bad_alloc (void)
 
   throw std::bad_alloc ();
 }
 
 void
 octave_rethrow_exception (void)
 {
   if (octave_interrupt_state)
-    octave_throw_interrupt_exception ();
+    {
+      octave_interrupt_state = -1;
+      octave_throw_interrupt_exception ();
+    }
   else
     {
       switch (octave_exception_state)
 	{
 	case octave_exec_exception:
 	  octave_throw_execution_exception ();
 	  break;
 
